oai_agents/common/population.py

```diff
@@ -147,16 +185,17 @@ def save_population(args, population):
         rt.save_agents(tag=KeyCheckpoints.MOST_RECENT_TRAINED_MODEL)
 
 
-def get_population(args,
-                   ck_rate,
-                   total_training_timesteps,
-                   train_types,
-                   eval_types,
-                   num_SPs_to_train,
-                   unseen_teammates_len=0,
-                   force_training=False,
-                   tag=KeyCheckpoints.MOST_RECENT_TRAINED_MODEL,
-                   ):
+def get_categorized_SP_population(
```

Comment:
What do you mean by 'categorized'?

Reply:
The function `get_population` was originally a clear name because it solely generated a population of agents. Now, its clarity is diminished because the code has been updated to support two functionalities: 

1. Generating a categorized population of agents (low, medium, and high performance).
2. Generating a population of uncategorized agents, supported by `Multi_Setup_Trainer`.

Since the latter functionality does not involve categorization, the name `get_population`, used by the first, may no longer fully represent the function's purpose.

Reply:
I'm confused by the name change as well. I don't see how this function is providing categorized agents. The role of [generate_TC()](https://github.com/HIRO-group/multiHRI/blob/main/oai_agents/common/teammates_collection.py#L186) is to separate the population into categories. Is that function unnecessary now?

Reply:
I rename it to get_performance_based_population_by_layouts 
This name may be too long, but it also provides the most description and is also pretty clear what this function really work for.